source/php/Integrations/Image/Image.php

```diff
@@ -0,0 +1,152 @@
+<?php 
+
+namespace ComponentLibrary\Integrations\Image;
+
+class Image implements ImageInterface {
+
+  private array $commonScreenSizes = [425, 768, 1024, 1440]; // Common screen sizes (width only) to generate sizes for
+
+  private mixed $resolver; // The callable that resolves the image in the native system
+
+  public function __construct(
+    private int $imageId, 
+    private array $imageSize, 
+    callable $resolver,
```

Comment:
Could we make the resolver even easier to implement by demanding something like a `ImageResolver` interface?

Reply:
I have intentionally omitted to require and implement a real interface for this. It seems a little more light weight to do it this way. It might be a bad idéa, but with this design we can resolve images with anonymous functions for example.

Reply:
Added Interface way.